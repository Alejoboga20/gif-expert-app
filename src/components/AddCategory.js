import React, { useState } from 'react';

const AddCategory = () => {
  const [inputValue, setInputValue] = useState('');

  const handleOnChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleOnSubmit = (e) => {
    e.preventDefault();
    setInputValue('');
  };

  return (
    <form onSubmit={handleOnSubmit}>
      <input type='text' value={inputValue} onChange={handleOnChange} />
    </form>
  );
};

export default AddCategory;
