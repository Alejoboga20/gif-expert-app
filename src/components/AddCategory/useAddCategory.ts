import { useState } from 'react';
import { FormElement } from '@nextui-org/react';

export const useAddCategory = ({ setCategories }: UseAddCategoryArgs) => {
	const [inputValue, setInputValue] = useState('');

	const handleOnChange = (e: React.ChangeEvent<FormElement>) => {
		const { value } = e.target;

		setInputValue(value);
	};

	const handleOnSubmit = (e: React.FormEvent<HTMLFormElement>) => {
		e.preventDefault();

		if (inputValue.trim().length > 2) {
			setCategories((cats) => [inputValue, ...cats]);
			setInputValue('');
		}
	};

	return {
		handleOnChange,
		inputValue,
		handleOnSubmit,
	};
};

interface UseAddCategoryArgs {
	setCategories: React.Dispatch<React.SetStateAction<string[]>>;
}
