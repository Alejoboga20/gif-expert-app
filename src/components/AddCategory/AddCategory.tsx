import { Input } from '@nextui-org/react';
import { styles } from './AddCategory.style';
import { useAddCategory } from './useAddCategory';

export const AddCategory = ({ setCategories }: AddCategoryProps) => {
	const { handleOnChange, inputValue, handleOnSubmit } = useAddCategory({ setCategories });

	return (
		<form onSubmit={(e) => handleOnSubmit(e)}>
			<Input
				className='input'
				labelPlaceholder='Gif'
				clearable
				type='text'
				value={inputValue}
				onChange={handleOnChange}
				css={styles.input}
			/>
		</form>
	);
};

interface AddCategoryProps {
	setCategories: React.Dispatch<React.SetStateAction<string[]>>;
}
