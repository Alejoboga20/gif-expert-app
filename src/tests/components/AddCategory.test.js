import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('AddCategory Tests', () => {
  const setCategories = () => {};
  const wrapper = shallow(<AddCategory setCategories={setCategories} />);

  test('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change text input', () => {
    const input = wrapper.find('input');
    const value = 'Hello World';
    input.simulate('change', { target: { value } });
  });
});
