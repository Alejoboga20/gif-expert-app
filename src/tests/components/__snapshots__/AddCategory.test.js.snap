// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory Tests should render properly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
